From scot@genroco.com Tue Jul 24 14:52:11 2001
Date: Tue, 24 Jul 2001 13:00:20 -0500
From: Scot W. Hetzel <scot@genroco.com>
To: Cyrus-Bugs <cyrus-bugs+@andrew.cmu.edu>
Subject: Cyrus-SASL Configure.in db3 detection on FreeBSD

I noticed a problem with the latest DB3 detection.  FreeBSD has the
DB 1.85 db.h file under /usr/include, which will satisfy the first
AC_CHECK_HEADER.  The Cyrus-SASL configure.in file should check for
db3/db.h first and then for db.h.

Also there is a missing AC_CHECK_LIB for db3, as FreeBSD's DB3 port
installs DB3 as libdb3.so.2.

There is also another problem with configure.in, when it first
attempts to check for krb_mk_priv, it fails on FreeBSD (no -lcom_err).
The second attempt adds -lcom_err, but it will also fail due to the
previous cached value from the first test.

I have attached a patch that will fix both of these problems in the
latest CVS sources.

Scot W. Hetzel

--- ../cyrus-sasl.orig/configure.in	Tue Jul 24 09:36:47 2001
+++ ../cyrus-sasl/configure.in	Tue Jul 24 11:48:47 2001
@@ -143,34 +143,35 @@
   dblib=auto_detect)
 SASL_DB_LIB=""
 
+dnl this is unbelievably painful due to confusion over what db-3 should be
+dnl named and where the db-3 header file is located.  arg.
+AC_DEFUN(BERKELEY_DB_CHK1,
+[	for dbname in db-3.2 db3.2 db-3.1 db3.1 db-3 db3 db
+	  do
+	    AC_CHECK_LIB($dbname, db_create, SASL_DB_LIB="-l$dbname";
+	      dblib="berkeley"; break, dblib="no")
+	  done
+	if test "$dblib" = "no"; then
+	  AC_CHECK_LIB(db, db_open, SASL_DB_LIB="-ldb";
+	    dblib="berkeley"; dbname=db,
+	    dblib="no")
+	fi
+])
 
 AC_DEFUN(BERKELEY_DB_CHK,
-[	AC_CHECK_HEADER(db.h,
-			AC_CHECK_LIB(db-3, db_create, SASL_DB_LIB="-ldb-3";
-			   dblib="berkeley",
-			AC_CHECK_LIB(db, db_create, SASL_DB_LIB="-ldb";
-			   dblib="berkeley",
-			AC_CHECK_LIB(db, db_open, SASL_DB_LIB="-ldb"; 
-			   dblib="berkeley",
-                        dblib="no"))),
-dnl FreeBSD puts it in a wierd place
-		AC_CHECK_HEADER(db3/db.h,
-			AC_CHECK_LIB(db-3, db_create, SASL_DB_LIB="-ldb-3";
-			   AC_DEFINE(HAVE_DB3_DB_H)
-			   dblib="berkeley",
-			AC_CHECK_LIB(db, db_create, SASL_DB_LIB="-ldb";
-			   AC_DEFINE(HAVE_DB3_DB_H)
-			   dblib="berkeley",
-			AC_CHECK_LIB(db, db_open, SASL_DB_LIB="-ldb"; 
-			   AC_DEFINE(HAVE_DB3_DB_H)
-			   dblib="berkeley",
-                        dblib="no"))),
-			dblib="no"))
+[	dnl FreeBSD puts it in a wierd place
+	AC_CHECK_HEADER(db3/db.h,
+			BERKELEY_DB_CHK1(),
+		AC_CHECK_HEADER(db.h,
+			BERKELEY_DB_CHK1()
+			if test "$dblib" = "berkeley"; then
+			  AC_DEFINE(HAVE_DB3_DB_H)
+			fi,
+			dblib="no"),
+			dblib="no")
 ])
 
 case "$dblib" in
-dnl this is unbelievably painful due to confusion over what db-3 should be
-dnl named.  arg.
   berkeley)
 	BERKELEY_DB_CHK()
 	;;
@@ -456,11 +456,15 @@
   fi
 
   if test "$with_des" != no; then
+    case "$host_os" in
+	freebsd*)
+	    COM_ERR="-lcom_err"
+	    ;;
+    esac
     AC_CHECK_HEADER(krb.h,
-      AC_CHECK_LIB(krb, krb_mk_priv, COM_ERR="",
-	AC_CHECK_LIB(krb, krb_mk_priv, COM_ERR="-lcom_err",
-                     AC_WARN(No Kerberos V4 found); krb4=no, -ldes -lcom_err),
-        -ldes),
+      AC_CHECK_LIB(krb, krb_mk_priv,:,
+		   AC_WARN(No Kerberos V4 found); krb4=no,
+		   -ldes $COM_ERR),
       AC_WARN(No Kerberos V4 found); krb4=no)
   else
     AC_WARN(No DES library found for Kerberos V4 support)
From scot@genroco.com Tue Jul 24 14:52:17 2001
Date: Tue, 24 Jul 2001 13:09:08 -0500
From: Scot W. Hetzel <scot@genroco.com>
To: Cyrus-Bugs <cyrus-bugs+@andrew.cmu.edu>
Subject: Re: Cyrus-SASL Configure.in db3 detection on FreeBSD

    [ The following text is in the "Windows-1252" character set. ]
    [ Your display is set for the "US-ASCII" character set.  ]
    [ Some characters may be displayed incorrectly. ]

From: "Scot W. Hetzel" <scot@genroco.com>
> +[ dnl FreeBSD puts it in a wierd place
> + AC_CHECK_HEADER(db3/db.h,
   BERKELEY_DB_CHK1()
   if test "$dblib" = "berkeley"; then
     AC_DEFINE(HAVE_DB3_DB_H)
   fi,

> + AC_CHECK_HEADER(db.h,
> + BERKELEY_DB_CHK1()

This needs to move up to the db3/db.h check
> + if test "$dblib" = "berkeley"; then
> +   AC_DEFINE(HAVE_DB3_DB_H)
> + fi,

> + dblib="no"),
> + dblib="no")
>  ])
>  


